### PR TITLE
Backend firebase initial setup

### LINKED ISSUE #34: Setup the database

### PR DESCRIPTION
**Pull request by MikolJiron**
Initialization for firebase.

### CODE REVIEW

**Review by Lorraine-Lyu — APPROVED** [review]

**Comment by MikolJiron** [comment]
@googlebot I consent.

File: src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java [hunk 1]
```diff
@@ -29,8 +33,18 @@
...
 @WebServlet("/college-data")
 public class CollegeServlet extends HttpServlet {
+  private Firestore database;
+  private Gson gson;
 
-  private final Gson gson = new Gson();
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    try {
+      database = FirestoreClient.getFirestore(FirebaseAppManager.getApp());
```
Comment (CarlosM10011) [inline src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java:42]: I think this method is called only when the servelet comes online, meaning that the credentials will only be initialized once. Have you considered moving it to doGet instead?

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java (hunk 1 above)
Comment (MikolJiron) [reply src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java:42]: Yeah that works.

**Comment by MikolJiron** [comment]
@googlebot I consent.

File: src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java [hunk 2]
```diff
@@ -34,20 +34,15 @@
...
 public class CollegeServlet extends HttpServlet {
   private Firestore database;
-  private Gson gson;
+  private Gson gson = new Gson();
 
   @Override
-  public void init(ServletConfig config) throws ServletException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     try {
       database = FirestoreClient.getFirestore(FirebaseAppManager.getApp());
```
Comment (CarlosM10011) [inline src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java:42]: Maybe get rid of class instance variable?

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java (hunk 2 above)
Comment (MikolJiron) [reply src/main/java/com/google/step/finscholar/servlets/CollegeServlet.java:42]: Yeah good idea.

**Review by CarlosM10011 — APPROVED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+   *   If the app initializes in production, we return credentials associated with the app.
+   *   Else, we use the default application credentials.
+   * @return - A GoogleCredentials object representing the credentials needed to start up the Firebase service.
+   */
+  private static GoogleCredentials getCredentials() throws IOException {
+    if(SystemProperty.environment.value() == SystemProperty.Environment.Value.Production) {
+      List<String> scopes =
+        Arrays.asList(
+          "https://www.googleapis.com/auth/cloud-platform",
+          "https://www.googleapis.com/auth/userinfo.email");
```
Comment (tinagrasso) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java]: Constants.

File: src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  private static FirebaseApp app = null;
+
+  /**
+   * This method initializes the Google Credentials needed to start up the Firebase service.
+   *   If the app initializes in production, we return credentials associated with the app.
+   *   Else, we use the default application credentials.
+   * @return - A GoogleCredentials object representing the credentials needed to start up the Firebase service.
+   */
+  private static GoogleCredentials getCredentials() throws IOException {
+    if(SystemProperty.environment.value() == SystemProperty.Environment.Value.Production) {
```
Comment (tinagrasso) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java]: Add a space between the if and paren.

File: src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+   * Initialize the Firebase app.
+   * @return - A FirebaseApp object that has been initialized to work with the current running build.
+   */
+  public static FirebaseApp getApp() throws IOException {
+    if(app == null) {
+      FirebaseOptions options =
+          new FirebaseOptions.Builder()
+              .setCredentials(getCredentials())
+              .setDatabaseUrl("https://viewing-step-2020-v2.firebaseio.com")
+              .setProjectId("viewing-step-2020-v2")
```
Comment (tinagrasso) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java]: Constants.

File: src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      return GoogleCredentials.getApplicationDefault();
+    }
+  }
+
+  /** 
+   * Initialize the Firebase app.
+   * @return - A FirebaseApp object that has been initialized to work with the current running build.
+   */
+  public static FirebaseApp getApp() throws IOException {
+    if(app == null) {
```
Comment (tinagrasso) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseAppManager.java]: Add a space.

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by hzimz — APPROVED** [review]

**Review by tinagrasso — APPROVED** [review]

**Pull request merged**